docs/features/creating_container.md

```diff
@@ -81,3 +81,59 @@ func TestIntegrationNginxLatestReturn(t *testing.T) {
 	}
 }
 ```
+
+## GenericReusableContainer
+
+`testcontainers.GenericReusableContainer` reuses a container if it exists or creates a generic container with parameters.
+Reusing will work only if you pass an existing container's name via 'req.Name' field.
+If the name is empty, or it is not in a list of existing containers, the function will create a new generic container
```

Comment:
Reading this description, which is pretty clear, this thought came to my mind: why adding a new type of container that could be used in a wrong manner, like creating a GenericReusableContainer but with a name that is not reusable?

This is thinking out loud: do you think adding the same logic to the reusable container into the generic container would make sense? Please take a look at how the Java folks do it: https://github.com/testcontainers/testcontainers-java/blob/de1324ed2800eff4da326d0c23d281399d006bc0/core/src/main/java/org/testcontainers/containers/GenericContainer.java#L384

The GenericContainer, if defined as reusable, reuses the container ID. I think this API would be simpler in terms of usability: as a consumer I don't have to remember the different types of Containers, simply configure the generic one.

Again, this is an opinion, as valid as any other, so please tell us your thoughts

Reply:
@mdelapenya I want to suggest next behavior: we add `Reuse` optional flag to `GenericContainerRequest`, because maybe a user doesn't want to reuse an existing container..

```go
c, err := GenericReusableContainer(ctx, GenericContainerRequest{
	ContainerRequest: ContainerRequest{
		Image:        "nginx:1.17.6",
		ExposedPorts: []string{"80/tcp"},
		WaitingFor:   wait.ForListeningPort("80/tcp"),
		Name:         reusableContainerName,
	},
	Started: true,
	Reuse: true,
})
```

And we will get next possible options:
1) `Reuse` is true. We will reuse an existing container or create a new one if it not exists . If we reuse an existing container, we will ignore flag `Started`, because the container is already started. 
2) `Reuse` is false. Try to create a new container or return an error if a container already exists.

Wdyt?

Reply:
I like it! Please go for it 🚀